server/src/main/java/org/opensearch/index/translog/TranslogWriter.java

```diff
@@ -438,8 +438,43 @@ public TranslogReader closeIntoReader() throws IOException {
         synchronized (syncLock) {
             try (ReleasableLock toClose = writeLock.acquire()) {
                 synchronized (this) {
+                    Long translogWithoutFooterChecksum = null;
+                    Long translogWithFooterChecksum = null;
                     try {
+                        if (header.getTranslogHeaderVersion() >= TranslogHeader.VERSION_WITH_FOOTER) {
+                            // If we are adding a footer, change the totalOffset.
+                            // This will ensure that footer length is included in the checkpoint.
+                            totalOffset += TranslogFooter.footerLength();
+                        }
+
                         sync(); // sync before we close..
+
+                        if (header.getTranslogHeaderVersion() >= TranslogHeader.VERSION_WITH_FOOTER) {
+                            // Post sync, we will add the footer to the translog.
+                            assert translogCheckedContainer != null : "checksum has not been calculated for the translog";
+                            // add footer to the translog file.
+                            // The checksum in the footer consists of header + body.
+                            byte[] footer = TranslogFooter.write(
+                                channel,
+                                translogCheckedContainer.getChecksum(),
+                                !Boolean.TRUE.equals(remoteTranslogEnabled)
```

Comment:
Doubt: as we have already written the translog to disk, after that we are writing footer to file. Why is it that we are forcing a sync again only in case of when `remoteTranslogEnabled` is `false`.?

Reply:
So presently Remote translog does not flush the changes to disk and only keeps it in memory. During sync workflow, it would write the data to file channel in memory and then during rollover, we will upload it directly to remote store.

On the other hand, for local translog, we are flushing the changes to disk on each sync. 

Now, in `TranslogFooter.write` method, we are doing the same operation- For local translog, we flush to disk and for remote translog, we write the data to file channel without flushing to disk.